platform-api/src/main/java/io/datacater/core/pipeline/PipelineEndpoint.java

```diff
@@ -112,20 +114,24 @@ public Uni<Response> deletePipeline(@PathParam("uuid") UUID uuid) {
   @Path("preview")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Uni<String> inspectStatic(String payload) {
+  public Uni<String> preview(String payload) {
+    LOGGER.debug(payload);
     HttpClient httpClient = HttpClient.newHttpClient();
-    Uni<NamedPod> namedPod = runnerPool.getStaticPod();
+    Uni<NamedPod> namedPod = runnerPool.getPod();
 
     return namedPod
-        .flatMap(
-            pod -> {
-              HttpRequest preview = pod.buildPost(payload, "/preview");
-              CompletableFuture<HttpResponse<String>> previewSend =
-                  httpClient.sendAsync(preview, BodyHandlers.ofString());
-
-              return Uni.createFrom().completionStage(previewSend);
-            })
-        .map(response -> response.body());
+        .onItem()
+        .transform(
+            Unchecked.function(
+                pod -> {
+                  HttpRequest preview = pod.buildPost(payload, "/preview");
+                  HttpResponse<String> previewSend =
+                      httpClient.send(preview, BodyHandlers.ofString());
```

Comment:
We should introduce a (configurable) timeout here, [similar to the one by the pipeline](https://github.com/DataCater/datacater/blob/main/pipeline/src/main/java/io/datacater/Pipeline.java#L53).

Reply:
Added without retry